src/librustc_middle/ty/layout.rs

```diff
@@ -2283,10 +2283,19 @@ impl<'tcx> ty::Instance<'tcx> {
     fn fn_sig_for_fn_abi(&self, tcx: TyCtxt<'tcx>) -> ty::PolyFnSig<'tcx> {
         let ty = self.monomorphic_ty(tcx);
         match ty.kind {
-            ty::FnDef(..) |
-            // Shims currently have type FnPtr. Not sure this should remain.
-            ty::FnPtr(_) => {
-                let mut sig = ty.fn_sig(tcx);
+            ty::FnDef(..) => {
+                // HACK(davidtwco/eddyb): This is a workaround for polymorphization considering
```

Comment:
Comma preferred to slash, this reads like namespacing to me.

Reply:
Fixed!